src/System.Memory/tests/Memory/Trim.cs

```diff
@@ -0,0 +1,200 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Xunit;
+
+namespace System.MemoryTests
+{
+    partial class MemoryTests
+    {
```

Comment:
The tests for each of `Memory`, `Span`, `RoM` and `RoS` are _exactly_ the same, save the types.

Reply:
Do we need to quadruplicate all the tests then?  Or could we just, for example, change it from:
```C#
Memory<int> memory = new Memory<int>(values).TrimStart(trim); 
Assert.True(System.Linq.Enumerable.SequenceEqual(expected, memory.ToArray()));
```
repeated with all of the inline datas four times, to instead be just once:
```C#
Assert.Equal<int>(expected, new Memory<int>(values).TrimStart(trim).ToArray());
Assert.Equal<int>(expected, new ReadOnlyMemory<int>(values).TrimStart(trim).ToArray());
Assert.Equal<int>(expected, new Span<int>(values).TrimStart(trim).ToArray());
Assert.Equal<int>(expected, new ReadOnlySpan<int>(values).TrimStart(trim).ToArray());
```
These are all effectively the same operation, just on variations of the data type, so I don't feel badly about having them all in the same test, and I prefer that much more than duplicating a dozen lines of InlineData for each one, where it's much more likely we'll end up accidentally evolving some of the tests but not all, and even if we were successful in keeping them in sync, it would require more coding effort and more review each time.

Reply:
@ahsonkhan, do you have an opinion?

Reply:
That's exactly how I had it in my private repo. Then realized that every other test in the official project was split 4 ways.